src/Raven.Server/Documents/Indexes/Persistence/Corax/QueryOptimizer/CoraxBooleanItem.cs

```diff
@@ -104,6 +108,92 @@ private CoraxBooleanItem(IndexSearcher searcher, Index index, FieldMetadata fiel
+                //         (double d, long l) => _indexSearcher.BetweenQuery(Field, d, Convert.ToDouble(l), leftSide: BetweenLeft, rightSide: BetweenRight, false, forwardIterator, streamingEnabled),
+                //         _ => throw new InvalidOperationException($"UnaryMatchOperation {Operation} is not supported for type {Term.GetType()}")
+                //     };
+                // }
+                // else
+                // {
+                //     baseMatch = (Operation, Term) switch
+                //     {
+                //         (UnaryMatchOperation.LessThan, long term) => _indexSearcher.LessThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.LessThan, double term) => _indexSearcher.LessThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.LessThan, string term) => _indexSearcher.LessThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //
+                //         (UnaryMatchOperation.LessThanOrEqual, long term) => _indexSearcher.LessThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.LessThanOrEqual, double term) => _indexSearcher.LessThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.LessThanOrEqual, string term) => _indexSearcher.LessThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //
+                //         (UnaryMatchOperation.GreaterThan, long term) => _indexSearcher.GreaterThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.GreaterThan, double term) => _indexSearcher.GreaterThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.GreaterThan, string term) => _indexSearcher.GreaterThanQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //
+                //
+                //         (UnaryMatchOperation.GreaterThanOrEqual, long term) => _indexSearcher.GreatThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.GreaterThanOrEqual, double term) => _indexSearcher.GreatThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         (UnaryMatchOperation.GreaterThanOrEqual, string term) => _indexSearcher.GreatThanOrEqualsQuery(Field, term, false, forwardIterator, streamingEnabled),
+                //         _ => throw new ArgumentException("This is only Greater*/Less* Query part")
+                //     };
+                // }
+                //
+                // return baseMatch;
+        }
```

Comment:
Leftovers or something that needs to be implemented yet?

Reply:
Meant to be implemented by @maciejaszyk , who took over this